Add scheduled check-in functionality

axif0 opened the pull request:
Add inputs for bot username and ignore label; update dependencies and configurations.

If we add [ignore-checkin](https://github.com/axif0/git-workflow/issues?q=state%3Aopen%20label%3A%22ignore-checkin%22), Issues will be skipped 

```
name: Check Inactive Issues
on:
  workflow_dispatch: # Allows manual triggering
  schedule:
    - cron: '0 0 * * *' # Optional: Runs daily
jobs:
  check-in:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Run Check in Bot
        uses: andrewtavis/test-check-in-bot@master
        with:
          repo-token: ${{ secrets.GITHUB_TOKEN }}
          days-inactive: '0.01' # 14.4 minutes for testing.
 
          comment-message: |
            <!-- AUTO-GENERATED CHECK-IN-message from check in bot-->
            {{ check-in-message }}
            
            Additionally, this issue has been inactive for {{ days-inactive }} days.
          bot-username: 'github-actions[bot]'
          ignore-label: 'ignore-checkin' 

```



Closes #3

andrewtavis commented:
Thanks for the great work here, @axif0! 😊 Is adding a label the only way we can cancel it, or would we be able to also have it check the comments in the issue and see if there's something like `checkin stop` and then that would prevent it from sending another comment?

axif0 commented:
> adding a label the only way we can cancel it

For now, only adding label is the only way to cancel it. But we can add `checkin stop`.

andrewtavis commented:
I'd say that that would be more intuitive than the labels. This is what I've seen in similar projects - i.e. stalebot 😊

axif0 commented:
Add stop-comment and skip-PR functionality to check-in bot

- Introduced new inputs: `stop-comment` (custom message), `skip-pr`, and `auto-add-label` to enhance bot behavior.
- Implemented logic to check for a stop comment and apply an ignore label if found.
- Added configuration logging for better visibility of bot settings. [see example action](https://github.com/axif0/git-workflow/actions/runs/14714013031/job/41293087471)
- Updated processing logic to allow skipping pull requests based on user preference.
- Enhanced summary reporting for processed items, including counts for comments and skips.


[Please see workflow file](https://github.com/axif0/git-workflow/blob/master/.github/workflows/test.yml)

andrewtavis reviewed (approved):
Changes look really good here, @axif0! I removed the `comment-message` argument in favor of just `check-in-message`. Not sure if there was a purpose of having a second message argument, but let's discuss in the issue!

I also added Yarn back in as the package manager in `package-lock.json` because I'm not able to build without it. Maybe there's an option to say that either Yarn or NPM is fine? I removed the `yarn.lock` file, but also maybe this is required. We'll figure the rest out 😊

Thanks for the great work here! 🚀
